packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRange.ts

```diff
@@ -147,13 +147,31 @@ export async function beaconBlocksMaybeBlobsByRange(
     }, [] as number[]);
 
     const dataColumnRequest = {...request, columns};
-    const [allBlocks, allDataColumnSidecars] = await Promise.all([
-      // TODO-das: investigate why partialDownload blocks is empty here
+    const allBlocksPromise =
       partialDownload && partialDownload.blocks.length > 0
-        ? partialDownload.blocks.map((blockInput) => ({data: blockInput.block}))
-        : network.sendBeaconBlocksByRange(peerId, request),
-      columns.length === 0 ? [] : network.sendDataColumnSidecarsByRange(peerId, dataColumnRequest),
-    ]);
+        ? Promise.resolve(partialDownload.blocks.map((blockInput) => ({data: blockInput.block})))
+        : network.sendBeaconBlocksByRange(peerId, request);
+
+    // DataColumnSidecars may take time for migration from unfinalized to finalized db
+    // So we retry to fetch the data column sidecars once with a delay
+    const allDataColumnSidecarsPromise =
+      columns.length === 0
+        ? Promise.resolve([])
+        : retry(() => network.sendDataColumnSidecarsByRange(peerId, dataColumnRequest), {retries: 2, retryDelay: 1000});
```

Comment:
why retry here?

Reply:
Explained in the comments above. When we request data columns the for finalized slots right at the moment we changed the epoch, those data side cars are not migrated from the hot to cold db yet. It take some fraction of time to migrate those. So here adding a single retry, give that fraction of time and we get those finalized data columns.

Reply:
I tend to agree.  We should not modify the primary code paths for e2e, we should modify the e2e to work with the code paths

Reply:
yes please, no logic changes in this PR just to pass the tests, the tests are passing when reverting https://github.com/ChainSafe/lodestar/pull/8206 without this modification and adding retry behavior

Reply:
That PR is fixing another issue when you come up with a block to migrate which have zero blobs. The changes here was more related to when we fetch sidecars which are not yet migrated from finalized to unfinalized state.